src/components/tasks/single/taskCard.tsx

```diff
@@ -1,27 +1,28 @@
-import React, { useState } from 'react';
+import React, { Dispatch, SetStateAction } from 'react';
 import { TaskCardData, TaskStatus } from '../../../models/task';
 import Card from '../../generic/card';
 import TaskCardContent from './taskCardContent';
-import TaskModal from './taskModal';
 import { getLevelToTextAndColor } from '../levelToTextAndColor';
 
 interface TaskCardProps {
     taskData: TaskCardData;
+    setTaskId: Dispatch<SetStateAction<number>>;
+    setModalIsOpen: Dispatch<SetStateAction<boolean>>;
 }
 
-const TaskCard = ({ taskData }: TaskCardProps) => {
+const TaskCard = ({ taskData, setTaskId, setModalIsOpen }: TaskCardProps) => {
     const { id, title, labels, status, level, points } = taskData;
     const { color, levelText } = getLevelToTextAndColor(level);
 
-    const [modalIsOpen, setModalIsOpen] = useState(false);
-
     return (
         <>
-            <TaskModal taskId={id} modalIsOpen={modalIsOpen} setModalIsOpen={setModalIsOpen} />
             <button
                 disabled={taskData.status === TaskStatus.BLOCKED}
                 className="w-full focus:outline-none"
-                onClick={() => setModalIsOpen(true)}
+                onClick={() => {
+                    setTaskId(id);
```

Comment:
I think it would be better to have only one function. You always pass the `taskid` and `true` to open modal. This could be simplified (TaskCardProps interface) also.